src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1079,24 +1131,92 @@ impl CheckValve {
 
 pub struct Reservoir {
     level_id: VariableIdentifier,
+    low_level_id: VariableIdentifier,
+
     max_capacity: Volume,
+    max_gaugeable: Volume,
     current_level: Volume,
     min_usable: Volume,
+
+    air_pressure: Pressure,
+
+    air_pressure_switches: Vec<PressureSwitch>,
+
+    level_switch: LevelSwitch,
+
+    leak_failure: Failure,
+    return_failure: Failure,
 }
 impl Reservoir {
-    const MIN_USABLE_VOLUME: f64 = 0.2; // Gallons
+    const MIN_USABLE_VOLUME_GAL: f64 = 0.2;
+
+    const LEAK_FAILURE_FLOW_GAL_PER_S: f64 = 0.1;
+
+    // Part of the fluid lost instead of returning to reservoir
+    const RETURN_FAILURE_LEAK_RATIO: f64 = 0.1;
 
     pub fn new(
         context: &mut InitContext,
         hyd_loop_id: &str,
         max_capacity: Volume,
+        max_gaugeable: Volume,
         current_level: Volume,
+        air_pressure_switches: Vec<PressureSwitch>,
+        low_level_threshold: Volume,
     ) -> Self {
+        let leak_failure = match hyd_loop_id {
+            "GREEN" => Failure::new(FailureType::GreenReservoirLeak),
+            "BLUE" => Failure::new(FailureType::BlueReservoirLeak),
+            "YELLOW" => Failure::new(FailureType::YellowReservoirLeak),
+            _ => Failure::new(FailureType::YellowReservoirLeak),
+        };
+
+        let return_failure = match hyd_loop_id {
+            "GREEN" => Failure::new(FailureType::GreenReservoirReturnLeak),
+            "BLUE" => Failure::new(FailureType::BlueReservoirReturnLeak),
+            "YELLOW" => Failure::new(FailureType::YellowReservoirReturnLeak),
+            _ => Failure::new(FailureType::YellowReservoirLeak),
+        };
```

Comment:
**issue**: `return_failure` is initialised with `FailureType::YellowReservoirLeak` in the `_` case.
**suggestion**: It's possible to reduce complexity/code in this area, see below. Note that `leak_failure` and `return_failure` can probably be inlined after this change.
```suggestion
        let leak_failure = Failure::new(FailureType::ReservoirLeak(hyd_loop_id));
        let return_failure = Failure::new(FailureType::ReservoirReturnLeak(hyd_loop_id));
```

**suggestion**: I'd consider adding a `HydraulicColour` enum type and using that throughout instead of using `String`.
```rust
enum HydraulicColour {
    Green,
    Blue
    Yellow,
}
```